src/main/java/com/microfocus/application/automation/tools/results/service/DefaultExternalEntityUploadServiceImpl.java

```diff
@@ -370,39 +370,49 @@ public void UploadExternalTestSet(AlmRestInfo loginInfo,
 		List<AlmTestSet> testsets = reportParserManager.parseTestSets(reportFilePath, testingFramework,  testingTool);
 
 		if(testsets == null) {
-			logger.log("Failed to parse file: " + reportFilePath);
+			logger.log("ERR: Failed to parse file: " + reportFilePath);
 			throw new ExternalEntityUploadException("Failed to parse file: " + reportFilePath);
 		} else  {
 			logger.log("INFO: parse resut file succeed.");
 		}
-		
-		if(testsets != null && testsets.size() >0 ) {
-			logger.log("INFO: Start to login to ALM Server.");
-			try {
-				if( restTool.login() ) {
-				
-					logger.log("INFO: Checking test folder...");
-					AlmTestFolder testFolder = createTestFolderPath(2, testFolderPath);
-					logger.log("INFO: Checking testset folder...");
-					AlmTestSetFolder testsetFolder = createTestSetFolderPath (0, testsetFolderPath);
-					if(testFolder != null && testsetFolder != null){
-						logger.log("INFO: Uploading ALM Entities...");
-						importExternalTestSet(
-								testsets, 
-								loginInfo.getUserName(), 
-								Integer.valueOf(testsetFolder.getId()), 
-								Integer.valueOf(testFolder.getId()), 
-								testingTool, 
-								subversion, 
-								jobName, 
-								buildUrl);
-					}
-				} else {
-					throw new ExternalEntityUploadException("Failed to login to ALM Server.");
-				}
-			} catch (Exception e) {
-				throw new ExternalEntityUploadException(e);
+
+		if (testsets.size() <= 0) {
+			logger.log("INFO: No testset to upload.");
+			return;
+		}
+		logger.log("INFO: Start to login to ALM Server.");
+
+		try {
+			if(!restTool.login()) {
+				throw new ExternalEntityUploadException("ERR: Failed to login to ALM Server.");
```

Comment:
Does someone catch this exception or it will be shown in the console output?

Reply:
Yes. It will be shown in the console output.

Reply:
Well. It seems the exception is swallowed out side. I will fix it.

Reply:
If the following exception occurs would you like to fail the build step? or at least mark the build step as a failure?